packages/components/src/popover/index.tsx

```diff
@@ -394,6 +436,7 @@ const Popover = (
 
 	const mergedFloatingRef = useMergeRefs( [
 		floating,
+		// @ts-expect-error dialogRef is returned by useDialog and seems correct, but somehow it's not picked up by TypeScript
```

Comment:
I need to potentially look more into why TypeScript doesn't seem to pick up the correct return value of `useDialog`

Reply:
I decided to type the `useDialog` hook, which removed the need for this `@ts-expect-error` comment